### PR TITLE
Fireloop

### PR DESCRIPTION
**Pull request by kattsushi**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<http://docs.fireloop.io/>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: fireloop/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+    /**
+     * 
+     * Boolean. Whether the method is static (eg. `MyModel.myMethod`). Use `false` to define the method on the prototype (for example, `MyModel.prototype.myMethod`). Default is true.
+     * default: true
+     */ 
+    isStatic?:	Boolean;
+
+    /**
+     *
+Additional notes, used by API documentation generators like Swagger.
```
Comment (andy-ms) [inline fireloop/index.d.ts]: Formatting

File: fireloop/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+}
+
+interface RemoteMethodArgument {
+    /**
+     * 	Argument name
+     */
+    arg:	String;
+    /**
+     * A text description of the argument. This is used by API documentation generators like Swagger.
+You can split long descriptions into arrays of strings (lines) to keep line lengths manageable.
```
Comment (andy-ms) [inline fireloop/index.d.ts]: Formatting

File: fireloop/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,595 @@
...
+"magna aliqua."
+] 
+```
+    */
+    description:	String | String[];
+    /**
+     * http	Object or Function	For input arguments: a function or an object describing mapping from HTTP request to the argument value. See HTTP mapping of input arguments below.
+     * http.target	
+     * Map the callback argument value to the HTTP response object. The following values are supported.
+* * status sets the res.statusCode to the provided value
```
Comment (andy-ms) [inline fireloop/index.d.ts]: Formatting

File: fireloop/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,595 @@
...
+     * // The login must be unique across all User instances.
+     * User.validatesUniquenessOf('login');
+     *
+     * // Assuming SiteUser.belongsTo(Site)
+     * // The login must be unique within each Site.
+     * SiteUser.validateUniquenessOf('login', { scopedTo: ['siteId'] });
+     * ```
+    
+    * @param {String} propertyName  Property name to validate.
+    * @options {Object} Options See below.
```
Comment (andy-ms) [inline fireloop/index.d.ts]: If this method takes an options object, express that in the signature, not in jsdoc. Don't need type declarations in the jsdoc if it's in the code.

**Review by ghost — COMMENTED** [review]
Be sure to lint!

**Comment by kattsushi** [comment]
Thanks

File: fireloop/2.0.0 [hunk 5]
```diff
@@ -0,0 +1 @@
+Terminate batch job (Y/N)? Terminate batch job (Y/N)? Terminate batch job (Y/N)?
```
Comment (andy-ms) [inline fireloop/2.0.0]: Delete

File: fireloop/index.d.ts [hunk 6]
```diff
@@ -360,10 +362,3 @@ interface RemoteMethodArgument {
     type: 'any' | 'Array' | 'boolean' | 'Buffer' | 'Date' | 'GeoPoint' | 'null' | 'number' | 'any' | 'string';
     default: string;
 }
-
-declare var ModelConstructor: ModelConstructor;
```
Comment (andy-ms) [inline fireloop/index.d.ts]: You didn't have to remove the ability to import it. The lint rule just expects you to remove the `declare module` wrapper, making it just:
```
declare var fireloop: ModelConstructor;
export = fireloop;
```

Changed the name to `fireloop` so it gets merged with the `fireloop` namespace. Then you can `import { Filter } from "fireloop";` if `Filter` is moved inside the `fireloop` namespace.

**Comment by DanielRosenwasser** [comment]
Hey @kattsushi - I'm a little confused. [It looks like Fireloop is actually written in TypeScript](https://github.com/mean-expert-official/fireloop., so is there a reason that it's not published with its own declaration files?

Additionally, [there is no npm package named fireloop](https://www.npmjs.com/package/fireloop), so I'm not entirely sure if this is currently named appropriately.

**Comment by kattsushi** [comment]
@DanielRosenwasser You're right, let me rethink this thank you very much

**Comment by andy-ms** [comment]
(Correct link: https://github.com/mean-expert-official/fireloop.io)

**Pull request closed without merging**